### PR TITLE
Feature/workflow test

### PR DESCRIPTION
**Pull request by Allen931**
Fix GitHub actions

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -49,13 +27,18 @@ jobs:
...
-          npm ci
-
+          cd root
+          npm run ci-all
+          
       - name: Test
-        run: npm test
+        run: |
+          cd root
+          npm run test-all
```
Comment (AustinZhu) [inline .github/workflows/node.js.yml:39]: You can just remove this step.

File: blog/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "analyze": "webpack --mode=production --env analyze",
     "lint": "eslint src --ext js,ts,tsx",
     "format": "prettier --write \"./**\"",
-    "test": "cross-env BABEL_ENV=test jest",
+    "test": "cross-env BABEL_ENV=test jest --passWithNoTests",
```
Comment (AustinZhu) [inline blog/package.json:12]: Instead of force npm to pass the test, you just need to remove the test phase in CI.

File: campus/package.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
     "analyze": "webpack --mode=production --env analyze",
     "lint": "eslint src --ext js,ts,tsx",
     "format": "prettier --write \"./**\"",
-    "test": "cross-env BABEL_ENV=test jest",
+    "test": "cross-env BABEL_ENV=test jest --passWithNoTests",
```
Comment (AustinZhu) [inline campus/package.json:12]: 1. What if someone writes a test and want to run it locally? -> He might find the test passes anyway.
2. What if later we have the test phase in CI? -> All these args need to be removed.

File: blog/src/root.component.test.tsx [hunk 4]
```diff
@@ -1,10 +0,0 @@
-import * as React from "react";
-import { render } from "@testing-library/react";
-import Root from "./root.component";
-
-describe("Root component", () => {
-  it("should be in the document", () => {
-    const { getByText } = render(<Root name="Testapp" />);
-    expect(getByText(/Testapp is mounted!/i)).toBeInTheDocument();
-  });
-});
```
Comment (AustinZhu) [inline blog/src/root.component.test.tsx:10]: Keep these dummy test files.

**Review by AustinZhu — CHANGES_REQUESTED** [review]
You can either:
1. keep the dummy test files and remove the test phase in CI.
2. keep the test phase in CI and make the condition in the test always true.

If you use `--passWithNoTests`, others might not be aware of that.
If you delete all the dummy test files, later people will need to create them from scratch.

**Comment by AustinZhu** [comment]
@Allen931 How's everything going?

**Comment by Allen931** [comment]
Added test cases. Sorry for being so late.

File: campus/package.json (hunk 3 above)
Comment (AustinZhu) [reply campus/package.json:12]: > 1. What if someone writes a test and want to run it locally? -> He might find the test passes anyway.
> 2. What if later we have the test phase in CI? -> All these args need to be removed.

@Allen931 Please DO NOT use --passWithNoTests

**Comment by AustinZhu** [comment]
> Added test cases. Sorry for being so late.

[Did you see this?](https://github.com/wasedatime/wasedatime-web/pull/219#discussion_r666094770)

**Comment by Allen931** [comment]
Sorry for the previous mistake, I thought I fix it.

**Comment by AustinZhu** [comment]
Can you use path in on_push event? Now even if someone makes a small change in a micro-app repo, all micro-apps need to be build.
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

**Pull request merged**